<div class="container">

<table style="width: 100%;"><tr>
<td>generate_ogrdb_report</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate OGRDB reports from specified files.</h2>

<h3>Description</h3>

<p>This creates the genotype report
(suffixed _ogrdb_report.csv) and the plot file (suffixed _ogrdb_plos.pdf).
Both are created in the directory holding the annotated read file, and the
file names are prefixed by the name of the annotated read file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">generate_ogrdb_report(
  ref_filename,
  inferred_filename,
  species,
  filename,
  chain,
  hap_gene,
  segment,
  chain_type,
  plot_unmutated,
  all_inferred = FALSE,
  format = "pdf"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ref_filename</code></td>
<td>
<p>Name of file containing IMGT-aligned reference genes in FASTA format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inferred_filename</code></td>
<td>
<p>Name of file containing sequences of inferred novel alleles, or '-' if none</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>species</code></td>
<td>
<p>Species name used in field 3 of the IMGT germline header with spaces omitted, if the reference file is from IMGT. Otherwise ‚Äù</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>Name of file containing annotated reads in AIRR, CHANGEO or IgDiscover format. The format is detected automatically</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chain</code></td>
<td>
<p>one of IGHV, IGKV, IGLV, IGHD, IGHJ, IGKJ, IGLJ, TRAV, TRAj, TRBV, TRBD, TRBJ, TRGV, TRGj, TRDV, TRDD, TRDJ</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hap_gene</code></td>
<td>
<p>The haplotyping columns will be completed based on the usage of the two most frequent alleles of this gene. If NA, the column will be blank</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segment</code></td>
<td>
<p>one of V, D, J</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chain_type</code></td>
<td>
<p>one of H, L</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_unmutated</code></td>
<td>
<p>Plot base composition using only unmutated sequences (V-chains only)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>all_inferred</code></td>
<td>
<p>Treat all alleles as novel</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>The format for the plot file ('pdf', 'html' or 'none')</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class="language-R"># prepare files for example
reference_set = system.file("extdata/ref_gapped.fasta", package = "ogrdbstats")
inferred_set = system.file("extdata/novel_gapped.fasta", package = "ogrdbstats")
repertoire = system.file("extdata/ogrdbstats_example_repertoire.tsv", package = "ogrdbstats")
file.copy(repertoire, tempdir())
repfile = file.path(tempdir(), 'ogrdbstats_example_repertoire.tsv')

generate_ogrdb_report(reference_set, inferred_set, 'Homosapiens',
          repfile, 'IGHV', NA, 'V', 'H', FALSE, format='none')

#clean up
outfile = file.path(tempdir(), 'ogrdbstats_example_repertoire_ogrdb_report.csv')
file.remove(repfile)
file.remove(outfile)
</code></pre>


</div>