<div class="container">

<table style="width: 100%;"><tr>
<td>example_rep</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Example repertoire data</h2>

<h3>Description</h3>

<p>A small example of the analytical datasets created by ogrdbstats from
repertoires and reference sets.
The dataset can be created by running the example shown for the function
read_input_data().
The dataset is created from example files provided with the package.
The repertoire data is taken from Rubelt et al. 2016,
&lt;doi: 10.1038/ncomms11112&gt;
</p>


<h3>Usage</h3>

<pre><code class="language-R">example_rep
</code></pre>


<h3>Format</h3>

<p>## 'example_rep' - a named list containing the following elements:
</p>

<table>
<tr>
<td style="text-align: left;">
ref_genes  </td>
<td style="text-align: left;"> named list of IMGT-gapped reference genes </td>
</tr>
<tr>
<td style="text-align: left;">
inferred_seqs </td>
<td style="text-align: left;"> named list of IMGT-gapped inferred (novel) sequences. </td>
</tr>
<tr>
<td style="text-align: left;">
input_sequences </td>
<td style="text-align: left;"> data frame with one row per annotated read, with
      CHANGEO-style column names. The column SEG_CALL is the gene call
     for the segment under analysis. Hence if segment is 'V', 'V_CALL' will
     be renamed 'SEG_CALL' whereas is segment is 'J', 'J_CALL' is renamed
     'SEG_CALL'. This simplifies downstream processing. Rows in the input
     file with ambiguous SEG_CALLs, or no call, are removed. </td>
</tr>
<tr>
<td style="text-align: left;">
genotype_db </td>
<td style="text-align: left;"> named list of gene sequences referenced in the annotated
     reads (both reference and novel sequences) </td>
</tr>
<tr>
<td style="text-align: left;">
haplo_details </td>
<td style="text-align: left;"> data used for haplotype analysis, showing allelic ratios
     calculated with various potential haplotyping genes </td>
</tr>
<tr>
<td style="text-align: left;">
genotype </td>
<td style="text-align: left;"> data frame containing information provided in the OGRDB
     genotype csv file </td>
</tr>
<tr>
<td style="text-align: left;">
calculated_NC </td>
<td style="text-align: left;"> a boolean that is TRUE if mutation counts were calculated
     by this library, FALSE if they were read from the annotated read file </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<h3>Source</h3>

<p>&lt;doi: 10.1038/ncomms11112&gt;
</p>


</div>