<div class="container">

<table style="width: 100%;"><tr>
<td>make_haplo_grobs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create haplotyping plots</h2>

<h3>Description</h3>

<p>Create haplotyping plots
</p>


<h3>Usage</h3>

<pre><code class="language-R">make_haplo_grobs(segment, haplo_details)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>segment</code></td>
<td>
<p>one of V, D, J</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>haplo_details</code></td>
<td>
<p>Data structure created by create_haplo_details</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>named list containing the following elements:
</p>

<table>
<tr>
<td style="text-align: left;">
    a_allele_plot </td>
<td style="text-align: left;">   plot showing allele usage for each potential haplotyping gene </td>
</tr>
<tr>
<td style="text-align: left;">
    haplo_grobs </td>
<td style="text-align: left;">     differential plot of allele usage for each usable haplotyping gene </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">haplo_grobs = make_haplo_grobs('V', example_rep$haplo_details)
</code></pre>


</div>