<div class="container">

<table style="width: 100%;"><tr>
<td>read_input_files</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read input files into memory</h2>

<h3>Description</h3>

<p>Read input files into memory
</p>


<h3>Usage</h3>

<pre><code class="language-R">read_input_files(
  ref_filename,
  inferred_filename,
  species,
  filename,
  chain,
  hap_gene,
  segment,
  chain_type,
  all_inferred
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ref_filename</code></td>
<td>
<p>Name of file containing IMGT-aligned reference genes in FASTA format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inferred_filename</code></td>
<td>
<p>Name of file containing sequences of inferred novel alleles, or '-' if none</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>species</code></td>
<td>
<p>Species name used in field 3 of the IMGT germline header with spaces omitted, if the reference file is from IMGT. Otherwise ‚Äù</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>Name of file containing annotated reads in AIRR, CHANGEO or IgDiscover format. The format is detected automatically</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chain</code></td>
<td>
<p>one of IGHV, IGKV, IGLV, IGHD, IGHJ, IGKJ, IGLJ, TRAV, TRAj, TRBV, TRBD, TRBJ, TRGV, TRGj, TRDV, TRDD, TRDJ</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hap_gene</code></td>
<td>
<p>The haplotyping columns will be completed based on the usage of the two most frequent alleles of this gene. If NA, the column will be blank</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segment</code></td>
<td>
<p>one of V, D, J</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chain_type</code></td>
<td>
<p>one of H, L</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>all_inferred</code></td>
<td>
<p>Treat all alleles as novel</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A named list containing the following elements:
</p>

<table>
<tr>
<td style="text-align: left;">
ref_genes  </td>
<td style="text-align: left;"> named list of IMGT-gapped reference genes </td>
</tr>
<tr>
<td style="text-align: left;">
inferred_seqs </td>
<td style="text-align: left;"> named list of IMGT-gapped inferred (novel) sequences. </td>
</tr>
<tr>
<td style="text-align: left;">
input_sequences </td>
<td style="text-align: left;"> data frame with one row per annotated read, with CHANGEO-style column names
                          One key point: the column SEG_CALL is the gene call for the segment under analysis. Hence if segment is 'V',
                          'V_CALL' will be renamed 'SEG_CALL' whereas is segment is 'J', 'J_CALL' is renamed 'SEG_CALL'. This simplifies
                          downstream processing.
                          Rows in the input file with ambiguous SEG_CALLs, or no call, are removed. </td>
</tr>
<tr>
<td style="text-align: left;">
genotype_db </td>
<td style="text-align: left;"> named list of gene sequences referenced in the annotated reads (both reference and novel sequences) </td>
</tr>
<tr>
<td style="text-align: left;">
haplo_details </td>
<td style="text-align: left;"> data used for haplotype analysis, showing allelic ratios calculated with various potential haplotyping genes </td>
</tr>
<tr>
<td style="text-align: left;">
genotype </td>
<td style="text-align: left;"> data frame containing information provided in the OGRDB genotype csv file </td>
</tr>
<tr>
<td style="text-align: left;">
calculated_NC </td>
<td style="text-align: left;"> a boolean that is TRUE if mutation counts were calculated by this library, FALSE if they were read from the annotated read file </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R"># Create the analysis data set from example files provided with the package
#(this dataset is also provided in the package as example_rep)
reference_set = system.file("extdata/ref_gapped.fasta", package = "ogrdbstats")
inferred_set = system.file("extdata/novel_gapped.fasta", package = "ogrdbstats")
repertoire = system.file("extdata/ogrdbstats_example_repertoire.tsv", package = "ogrdbstats")

example_data = read_input_files(reference_set, inferred_set, 'Homosapiens',
       repertoire, 'IGHV', NA, 'V', 'H', FALSE)
</code></pre>


</div>