<div class="container">

<table style="width: 100%;"><tr>
<td>make_barplot_grobs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create a barplot for each allele, showing number of reads distributed by mutation count</h2>

<h3>Description</h3>

<p>Create a barplot for each allele, showing number of reads distributed by mutation count
</p>


<h3>Usage</h3>

<pre><code class="language-R">make_barplot_grobs(
  input_sequences,
  genotype_db,
  inferred_seqs,
  genotype,
  segment,
  calculated_NC
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>input_sequences</code></td>
<td>
<p>the input_sequences data frame</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotype_db</code></td>
<td>
<p>named list of gene sequences in the personalised genotype</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inferred_seqs</code></td>
<td>
<p>named list of novel gene sequences</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotype</code></td>
<td>
<p>data frame created by calc_genotype</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segment</code></td>
<td>
<p>one of V, D, J</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>calculated_NC</code></td>
<td>
<p>a boolean, TRUE if mutation counts had to be calculated, FALSE otherwise</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>list of grobs
</p>


<h3>Examples</h3>

<pre><code class="language-R">barplot_grobs = make_barplot_grobs(
                      example_rep$input_sequences,
                      example_rep$genotype_db,
                      example_rep$inferred_seqs,
                      example_rep$genotype,
                      'V',
                      example_rep$calculated_NC
               )

</code></pre>


</div>