<div class="container">

<table style="width: 100%;"><tr>
<td>make_GXM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute gene expression correlation matrix.</h2>

<h3>Description</h3>

<p>This function computes a gene expression correlation matrix given a file of
transcript expression levels for each sample in the study. It returns a
correlation matrix with rownames and colnames as sample IDs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">make_GXM(expFile = NULL, gxmFilePrefix = NULL, idfile = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>expFile</code></td>
<td>
<p>Path to gene expression file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gxmFilePrefix</code></td>
<td>
<p>File path prefixes for outputting GCTA style binary
correlation matrices.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>idfile</code></td>
<td>
<p>Path to file containing family IDs and sample IDs with header FID and IID.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a correlation matrix of (N-samples x N-samples), with
rownames and colnames as sample IDs.
</p>


<h3>Examples</h3>

<pre><code class="language-R"> ## load gene expression values from vignette
 expressionFile &lt;- system.file(package = "OmicKriging",
                     "doc/vignette_data/ig_gene_subset.txt.gz")
 ## compute correlation matrix
 geneCorrelationMatrix &lt;- make_GXM(expressionFile)
</code></pre>


</div>