<div class="container">

<table style="width: 100%;"><tr>
<td>mltree</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Build Up a ML Tree</h2>

<h3>Description</h3>

<p>Given an alignment builds an ML tree.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mltree(msa, df = TRUE, gapl = TRUE, model = "WAG")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>msa</code></td>
<td>
<p>input alignment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>logical. When TRUE msa should be a dataframe, when FALSE msa should be a string giving the path to a fasta file containing the alignment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gapl</code></td>
<td>
<p>logical, when TRUE a gapless alignment is used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>allows to choose an amino acid models (see the function phangorn::as.pml)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function makes a NJ tree and then improvove it using an optimization procedure based on ML.
</p>


<h3>Value</h3>

<p>a ML optimized tree (and parameters)
</p>


<h3>See Also</h3>

<p>gapless_msa
</p>


<h3>Examples</h3>

<pre><code class="language-R">a &lt;- msa(sequences=c("RAPGT", "KMPGT", "ESGGT"), ids = letters[1:3])$ali
rownames(a) &lt;- letters[1:3]
tr &lt;- mltree(a)$tree
</code></pre>


</div>