<div class="container">

<table style="width: 100%;"><tr>
<td>studyVariants</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Retrieves all variants for a study.</h2>

<h3>Description</h3>

<p>For an input study ID, this function returns information of all variants across associated loci. The output also includes
information about the associated genes within the each loci.
</p>


<h3>Usage</h3>

<pre><code class="language-R">studyVariants(study_id)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>study_id</code></td>
<td>
<p>Character: Study ID(s) generated by Open Targets Genetics (e.g GCST90002357).</p>
</td>
</tr></table>
<h3>Value</h3>

<p>Returns a list of two data frames.
</p>
<p>the first data frame (tibble format) includes the loci data frame with following data structure:
</p>

<ul>
<li>
<p><code>variant.id</code>: <em>Character</em>. Variant ID.
</p>
</li>
<li>
<p><code>pval</code>: <em>Numeric</em>. P-value.
</p>
</li>
<li>
<p><code>variant.nearestCodingGene.symbol</code>: <em>Character</em>. Symbol of the nearest coding gene to the variant.
</p>
</li>
<li>
<p><code>variant.rsId</code>: <em>Character</em>. Variant rsID.
</p>
</li>
<li>
<p><code>variant.chromosome</code>: <em>Character</em>. Chromosome of the variant.
</p>
</li>
<li>
<p><code>variant.position</code>: <em>Integer</em>. Position of the variant.
</p>
</li>
<li>
<p><code>variant.nearestCodingGeneDistance</code>: <em>Integer</em>. Distance to the nearest coding gene.
</p>
</li>
<li>
<p><code>credibleSetSize</code>: <em>Integer</em>. Size of the credible set.
</p>
</li>
<li>
<p><code>ldSetSize</code>: <em>Integer</em>. Size of the LD set.
</p>
</li>
<li>
<p><code>oddsRatio</code>: <em>Numeric</em>. Odds ratio.
</p>
</li>
<li>
<p><code>beta</code>: <em>Numeric</em>. Beta value.
</p>
</li>
</ul>
<p>The second data frame includes gene information with following data structure:
</p>

<ul>
<li>
<p><code>score</code>: <em>Numeric</em>. Gene score.
</p>
</li>
<li>
<p><code>gene.id</code>: <em>Character</em>. Gene ID.
</p>
</li>
<li>
<p><code>gene.symbol</code>: <em>Character</em>. Gene symbol.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
result &lt;- studyVariants(study_id = "GCST003155")

## End(Not run)
</code></pre>


</div>