<div class="container">

<table style="width: 100%;"><tr>
<td>gwasColocalisation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Retrieve calculated GWAS colocalisation data</h2>

<h3>Description</h3>

<p>This function retrieves colocalisation data for a specific variant from a study with other GWAS studies.
It returns a data frame of the studies that colocalise with the input variant and study,
including details on the study and reported trait, index variant, and calculated coloc method (see Ref. below) outputs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gwasColocalisation(study_id, variant_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>study_id</code></td>
<td>
<p>Character: Open Target Genetics generated ID for the GWAS study.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variant_id</code></td>
<td>
<p>Character: Open Target Genetics generated ID for the variant (CHRPOSITION_REFALLELE_ALTALLELE or rsID).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a data frame of the studies that colocalise with the input variant and study. The table consists of the following data structure:
</p>

<ul>
<li>
<p><code>study.studyId</code>: <em>Character vector</em>. Study identifier.
</p>
</li>
<li>
<p><code>study.traitReported</code>: <em>Character vector</em>. Reported trait associated with the colocalisation.
</p>
</li>
<li>
<p><code>study.traitCategory</code>: <em>Character vector</em>. Trait category.
</p>
</li>
<li>
<p><code>indexVariant.id</code>: <em>Character vector</em>. Index variant identifier.
</p>
</li>
<li>
<p><code>indexVariant.position</code>: <em>Integer vector</em>. Index variant position.
</p>
</li>
<li>
<p><code>indexVariant.chromosome</code>: <em>Character vector</em>. Index variant chromosome.
</p>
</li>
<li>
<p><code>indexVariant.rsId</code>: <em>Character vector</em>. Index variant rsID.
</p>
</li>
<li>
<p><code>beta</code>: <em>Numeric vector</em>. Beta value associated with the colocalisation.
</p>
</li>
<li>
<p><code>h3</code>: <em>Numeric vector</em>. H3 value associated with the colocalisation.
</p>
</li>
<li>
<p><code>h4</code>: <em>Numeric vector</em>. H4 value associated with the colocalisation.
</p>
</li>
<li>
<p><code>log2h4h3</code>: <em>Numeric vector</em>. Log2 ratio of H4 to H3 values.
</p>
</li>
</ul>
<h3>References</h3>

<p>Giambartolomei, Claudia et al. “Bayesian test for colocalisation between pairs of genetic association studies using summary statistics.” PLoS genetics vol. 10,5 e1004383. 15 May. 2014, doi:10.1371/journal.pgen.1004383
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
colocalisation_data &lt;- gwasColocalisation(study_id = "GCST90002357", variant_id = "1_154119580_C_A")
colocalisation_data &lt;- gwasColocalisation(study_id = "GCST90002357", variant_id = "rs2494663")

## End(Not run)
</code></pre>


</div>