<div class="container">

<table style="width: 100%;"><tr>
<td>manhattan</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Retrieve GWAS summary statistics to create a Manhattan plot</h2>

<h3>Description</h3>

<p>This function retrieves summary statistics for a given GWAS study ID, which are used to generate a Manhattan plot.
The Manhattan plot is a graphical representation of genetic association studies, particularly in genome-wide association studies (GWAS).
It displays the results of statistical associations between genetic variants and a trait or disease of interest across the genome.
This function returns a data frame of the underlying data, which can be used to recreate the Manhattan plot using the <code>plot_manhattan()</code> function,
or for custom plots and downstream analysis.
</p>


<h3>Usage</h3>

<pre><code class="language-R">manhattan(study_id, pageindex = 0, pagesize = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>study_id</code></td>
<td>
<p>Character: Open Targets Genetics generated ID for the GWAS study.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pageindex</code></td>
<td>
<p>Int: Index of the current page (pagination index &gt;= 0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pagesize</code></td>
<td>
<p>Int: Number of records in a page (pagination size &gt; 0).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a data frame containing the Manhattan associations for the input study ID. The table consists of the following columns:
</p>

<ul>
<li>
<p><code>pval_mantissa</code>: Numeric vector. Mantissa of the p-value.
</p>
</li>
<li>
<p><code>pval_exponent</code>: Integer vector. Exponent of the p-value.
</p>
</li>
<li>
<p><code>credible_set_size</code>: Integer vector. Size of the credible set.
</p>
</li>
<li>
<p><code>ld_set_size</code>: Integer vector. Size of the LD set.
</p>
</li>
<li>
<p><code>total_set_size</code>: Integer vector. Total size of the set.
</p>
</li>
<li>
<p><code>pval</code>: Numeric vector. P-value.
</p>
</li>
<li>
<p><code>odds_ratio</code>: Logical vector. Odds ratio.
</p>
</li>
<li>
<p><code>odds_ratio_ci_lower</code>: Logical vector. Lower confidence interval of the odds ratio.
</p>
</li>
<li>
<p><code>odds_ratio_ci_upper</code>: Logical vector. Upper confidence interval of the odds ratio.
</p>
</li>
<li>
<p><code>beta</code>: Numeric vector. Beta value.
</p>
</li>
<li>
<p><code>beta_ci_lower</code>: Numeric vector. Lower confidence interval of the beta value.
</p>
</li>
<li>
<p><code>beta_ci_upper</code>: Numeric vector. Upper confidence interval of the beta value.
</p>
</li>
<li>
<p><code>direction</code>: Character vector. Direction of the effect.
</p>
</li>
<li>
<p><code>best_genes_score</code>: Numeric vector. Score of the best genes.
</p>
</li>
<li>
<p><code>best_genes_gene_id</code>: Character vector. Gene ID of the best genes.
</p>
</li>
<li>
<p><code>best_genes_gene_symbol</code>: Character vector. Gene symbol of the best genes.
</p>
</li>
<li>
<p><code>best_coloc_genes_score</code>: Numeric vector. Score of the best colocated genes.
</p>
</li>
<li>
<p><code>best_coloc_genes_gene_id</code>: Character vector. Gene ID of the best colocated genes.
</p>
</li>
<li>
<p><code>best_coloc_genes_gene_symbol</code>: Character vector. Gene symbol of the best colocated genes.
</p>
</li>
<li>
<p><code>best_locus2genes_score</code>: Numeric vector. Score of the best locus-to-genes.
</p>
</li>
<li>
<p><code>best_locus2genes_gene_id</code>: Character vector. Gene ID of the best locus-to-genes.
</p>
</li>
<li>
<p><code>best_locus2genes_gene_symbol</code>: Character vector. Gene symbol of the best locus-to-genes.
</p>
</li>
<li>
<p><code>variant_id</code>: Character vector. Variant ID.
</p>
</li>
<li>
<p><code>variant_position</code>: Integer vector. Variant position.
</p>
</li>
<li>
<p><code>variant_chromosome</code>: Character vector. Variant chromosome.
</p>
</li>
<li>
<p><code>variant_rs_id</code>: Character vector. Variant rsID.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
result &lt;- manhattan(study_id = "GCST90002357")
result &lt;- manhattan(study_id = "GCST90002357", pageindex = 2, pagesize = 50)

## End(Not run)
</code></pre>


</div>