<div class="container">

<table style="width: 100%;"><tr>
<td>plot_manhattan</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot results from <code>manhattan()</code>
</h2>

<h3>Description</h3>

<p>This function generates a Manhattan plot using the statistical summary data
obtained from the <code>manhattan()</code> function. Top 3 genes (based on p-value) are annotated per chromosome.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_manhattan(data)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>Data frame containing the necessary columns from <code>manhattan()</code> output for plotting:
- <code>variant_position</code>: Variant position
- <code>variant_chromosome</code>: Variant chromosome
- <code>pval</code>: P-value
- <code>variant_id</code>: Variant ID
- <code>best_locus2genes_score</code>: Best locus2genes score
- <code>best_locus2genes_gene_symbol</code>: Best locus2genes gene symbol</p>
</td>
</tr></table>
<h3>Value</h3>

<p>A Manhattan plot visualizing the GWAS results.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
p &lt;- manhattan(study_id = "GCST003044") %&gt;% plot_manhattan()

p

## End(Not run)
</code></pre>


</div>