<div class="container">

<table style="width: 100%;"><tr>
<td>gwasRegional</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Retrieve GWAS summary statistics for a genomic region.</h2>

<h3>Description</h3>

<p>For a given study ID and chromosomal region information, this function returns  data frame(tibble format) with all variants and their GWAS summary statistics.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gwasRegional(study_id, chromosome, start, end)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>study_id</code></td>
<td>
<p>Character: Open Target Genetics generated ID for the GWAS study.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosome</code></td>
<td>
<p>Character: Chromosome number as a string.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start</code></td>
<td>
<p>Integer: Start position of the specified chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>end</code></td>
<td>
<p>Integer: End position of the specified chromosome.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a data table of variant information and p-values with the following columns:
</p>

<ul>
<li>
<p><code>variant.id</code>: <em>Character vector</em>. Variant identifier.
</p>
</li>
<li>
<p><code>variant.chromosome</code>: <em>Character vector</em>. Chromosome of the variant.
</p>
</li>
<li>
<p><code>variant.position</code>: <em>Integer vector</em>. Position of the variant.
</p>
</li>
<li>
<p><code>pval</code>: <em>Numeric vector</em>. P-value.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
result &lt;- gwasRegional(study_id = "GCST90002357",
chromosome = "1", start = 153992685, end = 154155116)

## End(Not run)
</code></pre>


</div>