<div class="container">

<table style="width: 100%;"><tr>
<td>studyLocus2GeneTable</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Retrieve the locus-to-gene (L2G) data table for loci genes.</h2>

<h3>Description</h3>

<p>This function fetches the locus-to-gene (L2G) pipeline summary data table for
the neighboring genes of a variant in a GWAS study.
</p>


<h3>Usage</h3>

<pre><code class="language-R">studyLocus2GeneTable(study_id, variant_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>study_id</code></td>
<td>
<p>Character: Study ID(s) generated by Open Targets Genetics (e.g GCST90002357).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variant_id</code></td>
<td>
<p>Character: generated ID for variants by Open Targets Genetics (e.g. 1_154119580_C_A) or rsId (rs2494663).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a data frame with the summary statistics of the study and a data table containing various
calculated scores and features for any lead variant. The output table has the following data structure:
</p>

<ul>
<li>
<p><code>studyId</code>: <em>Character</em>. Study ID.
</p>
</li>
<li>
<p><code>variant.id</code>: <em>Character</em>. Variant ID.
</p>
</li>
<li>
<p><code>variant.rsId</code>: <em>Character</em>. Variant rsID.
</p>
</li>
<li>
<p><code>yProbaDistance</code>: <em>Numeric</em>. Distance score.
</p>
</li>
<li>
<p><code>yProbaModel</code>: <em>Numeric</em>. Model score.
</p>
</li>
<li>
<p><code>yProbaMolecularQTL</code>: <em>Numeric</em>. Molecular QTL score.
</p>
</li>
<li>
<p><code>yProbaPathogenicity</code>: <em>Numeric</em>. Pathogenicity score.
</p>
</li>
<li>
<p><code>yProbaInteraction</code>: <em>Numeric</em>. Interaction score.
</p>
</li>
<li>
<p><code>hasColoc</code>: <em>Logical</em>. Indicates if colocalization data is available.
</p>
</li>
<li>
<p><code>distanceToLocus</code>: <em>Numeric</em>. Distance to the locus.
</p>
</li>
<li>
<p><code>gene.id</code>: <em>Character</em>. Gene ID.
</p>
</li>
<li>
<p><code>gene.symbol</code>: <em>Character</em>. Gene symbol.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
result &lt;- studyLocus2GeneTable(study_id = "GCST90002357", variant_id = "1_154119580_C_A")
result &lt;- studyLocus2GeneTable(study_id = "GCST90002357", variant_id = "rs2494663")

## End(Not run)
</code></pre>


</div>