<div class="container">

<table style="width: 100%;"><tr>
<td>qtlColocalisationVariantQuery</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Retrieve  QTL colocalisation results of a variant</h2>

<h3>Description</h3>

<p>The colocalisation analysis in Open Target Genetics is performed using the coloc method (Giambartolomei et al., 2014).
Coloc is a Bayesian method which, for two traits, integrates evidence over all variants
at a locus to evaluate the following hypotheses:
- H0: No association with either trait
- H1: Association with trait 1, not with trait 2
- H2: Association with trait 2, not with trait 1
- H3: Association with trait 1 and trait 2, two independent SNPs
- H4: Association with trait 1 and trait 2, one shared SNP
This analysis tests whether two independent associations at the same locus are consistent
with having a shared causal variant. Colocalisation of two independent associations from two GWAS studies may suggest a shared causal mechanism.
</p>


<h3>Usage</h3>

<pre><code class="language-R">qtlColocalisationVariantQuery(study_id, variant_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>study_id</code></td>
<td>
<p>Character:  Study ID(s) generated by Open Targets Genetics (e.g GCST90002357).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variant_id</code></td>
<td>
<p>Character: generated ID for variants by Open Targets Genetics (e.g. 1_154119580_C_A) or rsId (rs2494663).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a data frame of the colocalisation information for a lead variant in a specific study.
The output is a tidy data frame with the following data structure:
</p>

<ul>
<li>
<p><code>qtlStudyName</code>: <em>Character vector</em>. QTL study name.
</p>
</li>
<li>
<p><code>phenotypeId</code>: <em>Character vector</em>. Phenotype ID.
</p>
</li>
<li>
<p><code>gene.id</code>: <em>Character vector</em>. Gene ID.
</p>
</li>
<li>
<p><code>gene.symbol</code>: <em>Character vector</em>. Gene symbol.
</p>
</li>
<li>
<p><code>name</code>: <em>Character vector</em>. Tissue name.
</p>
</li>
<li>
<p><code>indexVariant.id</code>: <em>Character vector</em>. Index variant ID.
</p>
</li>
<li>
<p><code>indexVariant.rsId</code>: <em>Character vector</em>. Index variant rsID.
</p>
</li>
<li>
<p><code>beta</code>: <em>Numeric</em>. Beta value.
</p>
</li>
<li>
<p><code>h4</code>: <em>Numeric</em>. h4 value.
</p>
</li>
<li>
<p><code>h3</code>: <em>Numeric</em>. h3 value.
</p>
</li>
<li>
<p><code>log2h4h3</code>: <em>Numeric</em>. Log2(h4/h3) value.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
result &lt;- qtlColocalisationVariantQuery(study_id = "GCST90002357", variant_id = "1_154119580_C_A")
result &lt;- qtlColocalisationVariantQuery(study_id = "GCST90002357", variant_id = "rs2494663")

## End(Not run)
</code></pre>


</div>